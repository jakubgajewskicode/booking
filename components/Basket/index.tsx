import {
  StyledSection,
  ButtonWrapper,
  Div,
  Title,
  Flex,
  SlotsWrapper,
  WorkerDetails,
} from './ui';
import { Button } from '../../utils/button';

const Basket = ({ data, setSendItemData, Slots, setAddSlotToArray }: any) => {
  return (
    <>
      <section>
        <Title>
          <h2>Your Basket</h2>
        </Title>
        {data.length >= 1 ? (
          <>
            <Flex>
              <WorkerDetails>
                {data.map((item: any) => (
                  <StyledSection key={item.id}>
                    <div>{item.name}</div>
                    <div>worker rating: {item.rating}</div>
                    {item.isNew && <div>New Employee</div>}
                    <div>worker ID: {item.id}</div>
                  </StyledSection>
                ))}
              </WorkerDetails>

              <SlotsWrapper>
                {Slots.map((slot: any) => (
                  <p key={slot}> Slot id: {slot}</p>
                ))}
              </SlotsWrapper>
            </Flex>

            <ButtonWrapper>
              <Button
                onClick={() => {
                  setSendItemData([]);
                  setAddSlotToArray([]);
                }}
              >
                Reset your order
              </Button>
            </ButtonWrapper>
          </>
        ) : (
          <Div>Nothing to show, add product to the basket</Div>
        )}
      </section>
    </>
  );
};

export default Basket;
