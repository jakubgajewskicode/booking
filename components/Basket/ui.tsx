import styled from 'styled-components';

const StyledSection = styled.section`
  display: flex;
  padding: 20px;
  padding: 20px 20px;
  justify-content: space-around;

  grid-row-start: 1;
`;

const ButtonWrapper = styled.div`
  width: 100%;
  text-align: center;
`;

const Div = styled.div`
  text-align: center;
`;

const Title = styled.div`
  text-align: center;
`;

const Flex = styled.div`
  display: flex;
`;

const SlotsWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  flex-direction: column;
  width: 20%;
`;

const WorkerDetails = styled.div`
  width: 80%;
`;

export {
  StyledSection,
  ButtonWrapper,
  Div,
  Title,
  Flex,
  SlotsWrapper,
  WorkerDetails,
};
