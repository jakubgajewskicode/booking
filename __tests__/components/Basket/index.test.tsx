import { shallow } from 'enzyme';

import Basket from '../../../components/Basket';

const data = {
  slots: [],
  setSendItemData: false,
  setItemId: 3,
  SlotsArray: [],
  setAddSlotToArray: false,
};

describe('Basket renders', () => {
  test('renders', () => {
    const basket = shallow(
      <Basket
        data={data.slots}
        setSendItemData={data.setSendItemData}
        setItemId={data.setItemId}
        SlotsArray={data.SlotsArray}
        setAddSlotToArray={data.setAddSlotToArray}
      />
    );
    const H2 = basket.find('h2');

    expect(basket).toMatchSnapshot();
    expect(H2).toHaveLength(1);
    expect(H2.text()).toEqual('Your Basket');
  });
});
