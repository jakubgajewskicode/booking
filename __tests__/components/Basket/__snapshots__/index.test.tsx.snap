// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket renders renders 1`] = `
<Fragment>
  <section>
    <ui__Title>
      <h2>
        Your Basket
      </h2>
    </ui__Title>
    <ui__Div>
      Nothing to show, add product to the basket
    </ui__Div>
  </section>
</Fragment>
`;
